oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,210 @@
+code/documentation to their own repo. For easier reintegration, Translation files will be
+kept in the same directory structure as the code/documentation they translate.
+
+Add GitHub Transifex App to openedx organization
+================================================
+
+Transifex's GitHub app will need to be added to the openedx GitHub organization in order
+to grant the app permissions to push/pull the Translation files. Currently, we manage the
+push/pull permissions for the edx-transifex-bot through a number of groups. The Transifex
+GitHub app is granted permission on a repository basis, and by moving all the
+Translations to a single repository we reduce the number of user groups we will have to
+manage.
+
+Connect i18n repository to Transifex via app
+============================================
+
+The Transifex webapp accepts configuration files for each Transifex project. By
+connecting the single repository containing all Translation files, we only need to make a
+single configuration file that allows the Transifex GitHub app to manage the Translation
+files. Based on the Translation Working Group's instruction, we can set parameters that
+automatically push and pull Translation files.
+
+Copy Transifex Memory
+=====================
+
+As a last step we can save all the progress the Open edX translators have accomplished by
+copying the Transifex Memory, the auto-translation feature that allows for Projects with
+similar strings to be automatically translated, from the old projects to this new one. By
+moving all the Translation Files to the same repository we can increase the reach of the
+Transifex Memory feature to help translate similar strings across the entire
+code/documentation base.
+
+Impacts
+*******
+
+Impact on Translators
+=====================
+
+As we approach the end of the translation upgrade process, we will need to tactically
+move from multiple transifex projects to a single project.  This will require
+coordination with our translators to ensure that moving forward they are providing
+translations in the right place.
+
+Impact on Site Operators
+========================
+
+Currently the translation files for any given service or library is stored at the same
+place as the code, this has generally simplified the deployment story in the past.  With
+this change, the translations files will move to their own repository.  As we deprecate
+the old translations files, the relevant deployment tooling will need to be updated to
+pull down the translations from the new repository as a part of the deployment process.
+This will impact both the old Ansible based tooling as well as any new docker based
+tooling.
+
+Impact on Developers
+====================
+
+While it won’t directly impact the day-to-day workflow of developers, due to the same
+reasons that we impact site operators(new translations location), we will have to update
+development tools as well.
```

Comment:
Will the process of creating a new repo be the same w/r/t setting up translations tooling/getting the integration working?

Reply:
We will need new instructions for how to enable translations for a new service/repo when it comes online.

Reply:
This has been updated.